internal/fullnode/node_key_builder_test.go

```diff
@@ -89,4 +90,40 @@ func TestBuildNodeKeySecrets(t *testing.T) {
 		require.NoError(t, err)
 		require.Empty(t, secrets)
 	})
+
+	t.Run("sets label for", func(t *testing.T) {
+		var crd cosmosv1.CosmosFullNode
+		crd.Spec.Replicas = 3
+
+		t.Run("type", func(t *testing.T) {
```

Comment:
Thank you for the tests. 

To me, the amount of test duplication is a signal for an abstraction. But I see why it was necessary given we want to test the public interface. 

I would probably abstract these into a test helper function.

Reply:
makes sense, I will push an update

Reply:
added HasTypeLabel assertion